neo/Persistence/Snapshot.cs

```diff
@@ -57,98 +59,41 @@ public virtual void Dispose()
         {
         }
 
-        private ECPoint[] _validators = null;
-        public ECPoint[] GetValidators()
+        public IEnumerable<(ECPoint PublicKey, BigInteger Votes)> GetRegisteredValidators()
         {
-            if (_validators == null)
+            byte[] script;
+            using (ScriptBuilder sb = new ScriptBuilder())
             {
-                _validators = GetValidators(Enumerable.Empty<Transaction>()).ToArray();
+                sb.EmitAppCall(NativeContract.NEO.ScriptHash, "getRegisteredValidators");
```

Comment:
Looks clear, @erikzhang.
Registered Validators are all all enrolled nodes?
When a node really enter as validator is restricted by a minimum number of votes `X`?